src/Castle.Core/DynamicProxy/Internal/AttributeUtil.cs

```diff
@@ -187,6 +187,26 @@ public static IEnumerable<CustomAttributeInfo> GetNonInheritableAttributes(this
 		}
 
 		public static IEnumerable<CustomAttributeInfo> GetNonInheritableAttributes(this ParameterInfo parameter)
+		{
+			return parameter.GetCustomAttributesToBeReproduced(ignoreInheritance: false);
+			//                                                 ^^^^^^^^^^^^^^^^^^^^^^^^
+			// The consequence of this is that custom attributes with `[AttributeUsage(Inherited = true)]` are not
+			// returned, because it is assumed that `parameter` will be able to inherit the custom attribute from
+			// a base type's corresponding method parameter.
+		}
+
+		internal static IEnumerable<CustomAttributeInfo> GetCustomAttributesToBeReproducedOnProxyCtor(this ParameterInfo parameter)
```

Comment:
(Let's keep this `internal` in order to not expand the public API surface unnecessarily).